modules/data-streams/src/javaRestTest/java/org/elasticsearch/datastreams/logsdb/qa/StandardVersusLogsIndexModeChallengeRestIT.java

```diff
@@ -39,85 +44,71 @@
 import java.util.List;
 import java.util.Map;
 
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThan;
+
 public class StandardVersusLogsIndexModeChallengeRestIT extends AbstractChallengeRestTest {
+    private final DataGenerator dataGenerator;
 
     public StandardVersusLogsIndexModeChallengeRestIT() {
         super("standard-apache-baseline", "logs-apache-contender", "baseline-template", "contender-template", 101, 101);
+        this.dataGenerator = new DataGenerator(
+            DataGeneratorSpecification.builder()
+                // Nested fields don't work with subobjects: false.
+                .withNestedFieldsLimit(0)
+                // TODO increase depth of objects
+                // Currently matching fails because in synthetic source all fields are flat (given that we have subobjects: false)
```

Comment:
Where is this set? I think we don't want to proceed with this..

Reply:
This is just a consequence of how synthetic source works. We'll get a synthetic field loader for every mapper and let them populate the synthetic source. With `subobjects: false` we only have leaf mappers - e.g. a keyword field `a.b.c` which will write `a.b.c: "foo"` in the source. It is object mappers who usually add structure to synthetic source and in this case there are no object mappers.

Reply:
We would have to write a special implementation of top-level synthetic field loader if we wanted to fix this.

Reply:
I was referring to using something like https://github.com/elastic/elasticsearch/pull/110524, instead of `subobjects: false`.. We know that the latter has shortcomings, shouldn't be used in logsdb mode in its current form..

Reply:
Ah yes, i definitely think that we should use `auto` once it is available. Just to clarify - do you propose to remove `subobjects: false` until then? I am okay with that.

Reply:
Right, we should probably do so if the matching logic can tolerate it. Once `auto` is in, it should apply to both standard and logsdb as it's orthogonal.